### PR TITLE
Named various fields mostly related to undead

### PR DESCRIPTION
**Pull request by AtomicChicken**
As per the title, alongside some miscellaneous stuff. I don't yet know how to properly format structures beyond giving them names and leaving comments. As such, I've tried to make the comments as detailed as possible to aid whoever gets around to formally editing this. Please let me know if anything's unclear.

### CODE REVIEW

File: df.history.xml [hunk 1]
```diff
@@ -243,11 +243,11 @@
...
             <stl-vector type-name='int32_t' name="unk_c0"/>
-            <int32_t name="unk_d0"/>
-            <int32_t name="unk_d4"/>
-            <int32_t name="unk_d8"/>
-            <int32_t name="unk_dc"/>
-            <stl-string name="unk_e0"/>
+            <int32_t name="original_histfig_id" ref-target='historical_figure' comment='ID of the historical figure who was reanimated to produce undead historical figure'/>
+            <int32_t name="original_race" ref-target='creature_raw' comment='race of the historical figure who was reanimated to produce undead historical figure'/>
+            <int32_t name="original_caste" ref-target='caste_raw' aux-value='$$.original_race' comment='caste of the historical figure who was reanimated to produce undead historical figure'/>
+            <int32_t name="root_body_part_id" comment='ID of the root body part in the corpse or corpse piece which was reanimated to produce undead historical figure'/>
```
Comment (AtomicChicken) [inline df.history.xml]: not sure how to reference body part

File: df.history.xml [hunk 2]
```diff
@@ -243,11 +243,11 @@
...
             <stl-vector type-name='int32_t' name="unk_b0"/>
             <stl-vector type-name='int32_t' name="unk_c0"/>
-            <int32_t name="unk_d0"/>
-            <int32_t name="unk_d4"/>
-            <int32_t name="unk_d8"/>
-            <int32_t name="unk_dc"/>
-            <stl-string name="unk_e0"/>
+            <int32_t name="original_histfig_id" ref-target='historical_figure' comment='ID of the historical figure who was reanimated to produce undead historical figure'/>
+            <int32_t name="original_race" ref-target='creature_raw' comment='race of the historical figure who was reanimated to produce undead historical figure'/>
+            <int32_t name="original_caste" ref-target='caste_raw' aux-value='$$.original_race' comment='caste of the historical figure who was reanimated to produce undead historical figure'/>
```
Comment (AtomicChicken) [inline df.history.xml]: Not sure about the aux-value; is this correct?

Comment (lethosor) [reply df.history.xml]: Not sure. ref-target is the only one I really care about.

File: df.history.xml (hunk 1 above)
Comment (lethosor) [reply df.history.xml]: Me neither, it's probably fine without it

**Pull request merged**